### PR TITLE
Merged Parameter Modal and Test Parameter Modal. Added panel to show error

### PR DESCRIPTION
**Pull request by hemny-singh**

### CODE REVIEW

File: squealy/squealy-web/src/Components/temp/ApiDesignView.js [hunk 1]
```diff
@@ -27,13 +27,15 @@ export default class ApiDesignView extends Component {
...
-            testParameters={chart.testParameters}
+          <QueryEditor
+            query={chart.query}
+            parameters={chart.parameters}
             selectedChartChangeHandler={selectedChartChangeHandler} />
         </div>
         <div className="col-md-12">
-          <ResultSection 
+          <ResultSection
+            isError={chart.apiError}
```
Comment (swapnil95) [inline squealy/squealy-web/src/Components/temp/ApiDesignView.js]: Can these two props (isError and errorMessage) be merged into a single prop. For example isError could be a derived conclusion depending on the value of errorMessage prop

File: squealy/squealy-web/src/Utils.js [hunk 2]
```diff
@@ -400,3 +403,13 @@ export function isJsonString(str) {
     }
     return true;
 }
+
+
+export function checkObjectAlreadyExists (data, keyName, value) {
+  for (let i = 0; i < data.length; i++) {
```
Comment (devashishsharma2302) [inline squealy/squealy-web/src/Utils.js]: for loop should be replaced with a map function.

Comment (hemny-singh) [inline squealy/squealy-web/src/Utils.js]: We don't need a map here as I need this to be terminate the loop once the condition is true. Map will create new array with the results of I am returning. So `for` loop suits better here.

Comment (devashishsharma2302) [inline squealy/squealy-web/src/Utils.js]: Oh okay.  Good.

**Pull request merged**